olive/workflows/run/run.py

```diff
@@ -4,14 +4,32 @@
 # --------------------------------------------------------------------------
 import json
 import logging
+import subprocess
 from pathlib import Path
 from typing import Union
 
+from olive.systems.common import Device, SystemType
 from olive.workflows.run.config import RunConfig
 
 logger = logging.getLogger(__name__)
 
 
+DEPENDENCY_MAPPING = {
+    "device": {
+        SystemType.AzureML: ["azure-ai-ml>=0.1.0b6", "azure-identity"],
+        SystemType.Docker: ["docker"],
+        SystemType.Local: {Device.CPU: ["onnxruntime==1.13.1"], Device.GPU: ["onnxruntime-gpu==1.13.1"]},
+    },
+    "pass": {
+        "OnnxFloatToFloat16": ["onnxconverter-common"],
+        "OrtPerfTuning": ["psutil"],
+        "QuantizationAwareTraining": ["pytorch-lightning"],
+        "OpenVINOConversion": ["openvino==2022.3.0", "openvino-dev[tensorflow,onnx]==2022.3.0"],
+        "OpenVINOQuantization": ["openvino==2022.3.0", "openvino-dev[tensorflow,onnx]==2022.3.0"],
+    },
+}
```

Comment:
could we manage only one extra for these dependencies? from the time being, both here and the setup.py maintain the dependencies.

Reply:
put extra dependencies together in extra_dependencies.json